invenio_stats/aggregations.py

```diff
@@ -263,52 +266,67 @@ def _get_oldest_event_timestamp(self):
             return None
         return parser.parse(result[0]["timestamp"])
 
-    def agg_iter(self, lower_limit, upper_limit):
+    def _split_date_range(self, lower_limit, upper_limit):
+        res = {}
+        current_interval = lower_limit
+        delta = INTERVAL_DELTAS[self.interval]
+        while current_interval < upper_limit:
+            dt_key = current_interval.strftime(SUPPORTED_INTERVALS[self.interval])
+            res[dt_key] = current_interval
+            current_interval += delta
+
+        dt_key = upper_limit.strftime(SUPPORTED_INTERVALS[self.interval])
+        res[dt_key] = upper_limit
+        return res
+
+    def agg_iter(self, dt):
         """Aggregate and return dictionary to be indexed in ES."""
-        aggregation_data = {}
-        start_date = format_range_dt(lower_limit, self.interval)
-        end_date = format_range_dt(upper_limit, self.interval)
+        rounded_dt = format_range_dt(dt, self.interval)
         agg_query = dsl.Search(using=self.client, index=self.event_index).filter(
-            "range", timestamp={"gte": start_date, "lte": end_date}
+            "range", timestamp={"gte": rounded_dt, "lte": rounded_dt}
         )
         self.agg_query = agg_query
 
         # apply query modifiers
         for modifier in self.query_modifiers:
             self.agg_query = modifier(self.agg_query)
 
-        # TODO: remove histogram bucket
-        histogram = self.agg_query.aggs.bucket(
-            "histogram", "date_histogram", field="timestamp", interval=self.interval
-        )
-        bucket_size = get_bucket_size(
+        total_buckets = get_bucket_size(
             self.client,
             self.event_index,
             self.field,
-            start_date=start_date,
-            end_date=end_date,
+            start_date=rounded_dt,
+            end_date=rounded_dt,
         )
-        if bucket_size > 0:
-            terms = histogram.bucket(
-                "terms", "terms", field=self.field, size=bucket_size
-            )
-        else:
-            terms = histogram.bucket(
+
+        num_partitions = max(
```

Comment:
minor: is `max` needed? ceil would already bump it to 1 if it were `0.x`.

```python
math.ceil(0.1)
1
```

Reply:
I think this is a safety thing in case `total_buckets` is 0, i.e. when there are no events to be aggregated, in which case we need it, so that the whole aggregation task doesn't explode.

Reply:
Ah, it makes sense. Thanks!